src/main/java/org/isf/usergroups/rest/UserGroupController.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Open Hospital (www.open-hospital.org)
+ * Copyright © 2006-2024 Informatici Senza Frontiere (info@informaticisenzafrontiere.org)
+ *
+ * Open Hospital is a free and open source software for healthcare data management.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * https://www.gnu.org/licenses/gpl-3.0-standalone.html
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+package org.isf.usergroups.rest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.isf.usergroups.dto.UserGroupDTO;
+import org.isf.menu.manager.UserBrowsingManager;
+import org.isf.usergroups.mapper.UserGroupMapper;
+import org.isf.menu.model.UserGroup;
+import org.isf.permissions.dto.PermissionDTO;
+import org.isf.permissions.manager.GroupPermissionManager;
+import org.isf.permissions.manager.PermissionManager;
+import org.isf.permissions.mapper.PermissionMapper;
+import org.isf.permissions.model.GroupPermission;
+import org.isf.permissions.model.Permission;
+import org.isf.shared.exceptions.OHAPIException;
+import org.isf.utils.exception.OHDataValidationException;
+import org.isf.utils.exception.OHServiceException;
+import org.isf.utils.exception.model.OHExceptionMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
```

Comment:
The import order doesn't match project standards; see https://github.com/informatici/openhospital-api/blob/develop/.ide-settings/eclipse/OpenHospital.importorder

6=
5=com
4=org
3=jakarta
2=javax
1=java